src/source/src/table.rs

```diff
@@ -94,9 +94,10 @@ impl TableDmlHandle {
                 )));
             }
             let len = guard.changes_txs.len();
+            // Use session id instead of txn_id to choose channel so that we can preserve transaction order in the same session.
```

Comment:
Ah, it might be hard to say. Only hold if there's no scaling on the table.

Reply:
That's true. Let me add this to the comment.